js/__tests__/eventListenerCallbacks.js

```diff
@@ -10,10 +10,29 @@
 /* eslint-env jest */
 
 import NetInfo from '../index';
-import {NetInfoEventEmitter} from '../nativeInterface';
+import {RNCNetInfo, NetInfoEventEmitter} from '../nativeInterface';
 
 describe('react-native-netinfo', () => {
   describe('Event listener callbacks', () => {
+    beforeEach(() => {
+      NetInfo.clearEventListeners();
+
+      RNCNetInfo.getCurrentConnectivity.mockResolvedValue({
+        connectionType: 'cellular',
+        effectiveConnectionType: '3g',
+      });
+    });
+
+    it('should call the listener on listening', done => {
+      const listener = jest.fn();
+      NetInfo.addEventListener('connectionChange', listener);
+
+      setImmediate(() => {
+        expect(listener).toBeCalled();
+        done();
+      });
+    });
+
     it('should call the listener when the native event is emmitted', () => {
```

Comment:
```suggestion
    it('should call the listener when the native event is emitted', () => {
```

Not part of this changelist but saw it so thought I'd drop it as a suggestion.